crates/red_knot_python_semantic/resources/mdtest/unreachable.md

```diff
@@ -218,26 +255,24 @@ def f():
 
 In the example below, since we use `x` in the `inner` function, we use the "public" type of `x`,
 which currently refers to the end-of-scope type of `x`. Since the end of the `outer` scope is
-unreachable, we treat `x` as if it was not defined. This behavior can certainly be improved.
+unreachable, we need to make sure that we do not emit an `unresolved-reference` diagnostic:
 
 ```py
 def outer():
     x = 1
 
     def inner():
-        return x  # Name `x` used when not defined
+        reveal_type(x)  # revealed: Unknown
     while True:
         pass
 ```
 
-## No diagnostics in unreachable code
-
-In general, no diagnostics should be emitted in unreachable code. The reasoning is that any issues
-inside the unreachable section would not cause problems at runtime. And type checking the
-unreachable code under the assumption that it *is* reachable might lead to false positives:
+### Global constants
 
 ```py
-FEATURE_X_ACTIVATED = False
+from typing import Literal
+
+FEATURE_X_ACTIVATED: Literal[False] = False
```

Comment:
We don't support `Final` yet, so I needed to declare this as `Literal[False]`. Otherwise, the public type (which we use below) would be `Unknown | Literal[False]`.